src/index.js

```diff
@@ -197,10 +197,16 @@ class Libp2p extends EventEmitter {
 
     // Create Metrics
     if (this._options.metrics.enabled) {
-      this.metrics = new Metrics({
-        ...this._options.metrics,
-        connectionManager: this.connectionManager
+      const metrics = new Metrics({
+        ...this._options.metrics
       })
+
+      // listen for peer disconnect events
+      this.connectionManager.on('peer:disconnect', (connection) => {
+        metrics.onPeerDisconnected(connection.remotePeer)
```

Comment:
shouldn't be the connection manager to trigger a metrics change instead of an explicit `onPeerDisconnected` method?

Reply:
There's quite a bit of logic in the `onPeerDisconnected` method, it'd be better to keep it out of the connection manager, if that's what you mean.

That said I'd rather there was a generic event bus that the metrics collector could listen to, then it could listen for `peer:*` events and count connections without the connection manager knowing anything about metrics at all.